### PR TITLE
Resolved issues in PTF framework and enabled PTF Tests

### PR DESCRIPTION
**Pull request by sureshmarikkannu**
Following changes are added as part of this commit,
 - Upgraded TREX version to use 2.92-scapy-2.4.5
 - Resolved Trex port readiness issue during startup
 - Updated README with detailed procedure on how to configure PTF

### CODE REVIEW

File: ptf/config/4-ports-with-l2.yaml
Comment (gab-arrobo) [inline ptf/config/4-ports-with-l2.yaml:1]: Should we rename this file to make its name more generic?

Comment (sureshmarikkannu) [reply ptf/config/4-ports-with-l2.yaml:1]: done

**Review by gab-arrobo — COMMENTED** [review]
I am able to successfully run all of the `unary` tests, the `linerate baseline` test and the `linerate mbr` tests. However, the `linerate qos_metrics` test fails (TRex crashes because mbuf_64 is too small). I had to make a few changes to be able to run all of the tests. Were you able to run all of the tests without any changes?

File: ptf/Dockerfile [hunk 1]
```diff
@@ -45,8 +45,9 @@ ARG TREX_LIBS
 
 ENV TREX_SCRIPT_DIR=/trex-core-${TREX_VER}/scripts
 
-RUN wget https://github.com/stratum/trex-core/archive/${TREX_VER}.zip && \
-    unzip -qq ${TREX_VER}.zip && \
+RUN apk update && apk add --no-cache -U wget
+RUN wget https://github.com/stratum/trex-core/archive/refs/heads/v${TREX_VER}.zip && \
```
Comment (gab-arrobo) [inline ptf/Dockerfile:49]: Why do you have to make this change? Why not just something like this:
```suggestion
RUN wget https://github.com/stratum/trex-core/archive/v${TREX_VER}.zip && \
```

File: ptf/config/trex-cfg-for-ptf.yaml
Comment (gab-arrobo) [inline ptf/config/trex-cfg-for-ptf.yaml:1]: In the directory where this file is, there is also a couple files (`docker_setup.sh` and `upf.jsonc`), are these files used for something? If not, what about deleting them and move the `trex-cfg-for-ptf.yaml` to the `ptf` directory?

File: ptf/README.md [hunk 2]
```diff
@@ -74,8 +74,15 @@ hosting TRex, where results are asserted.
 ## Steps to run tests
 The run script assumes that the TRex daemon server and the UPF
 instance are already running on their respective machines. It also
-assumes that all config files in `config/` are configured correctly to
-route traffic to the UPF.
+assumes that all the following config files are configured correctly to
+route traffic to the UPF and vice versa.
+* `upf/scripts/docker_setup.sh` script updated with proper values for
```
[the remark this answers is not in the document]
Comment (sureshmarikkannu) [reply ptf/README.md:79]: It is not necessary to keep this repo in trex machine as well. We can keep only in upf and just configure trex IP. So keep in mind that, i just added this as a reference to setup UPF.

File: ptf/tests/linerate/baseline.py
[the remark this answers is not in the document]
Comment (sureshmarikkannu) [reply ptf/tests/linerate/baseline.py:1]: yes, i have updated it and verified both.

File: ptf/Dockerfile (hunk 1 above)
Comment (sureshmarikkannu) [reply ptf/Dockerfile]: done

File: ptf/config/trex-cfg-for-ptf.yaml
Comment (sureshmarikkannu) [reply ptf/config/trex-cfg-for-ptf.yaml:1]: These files are placed here for reference. i.e how to configured UPF related files for successfully setting up PTF tests. This contains working version of config/scripts for running PTF tests successfully. So it is used for reference.

File: ptf/config/docker_setup.sh [hunk 3]
```diff
@@ -202,3 +259,6 @@ docker run --name bess-routectl -td --restart unless-stopped \
 	--net container:pause --pid container:bess \
 	--entrypoint /route_control.py \
 	upf-epc-bess:"$(<VERSION)" -i "${ifaces[@]}"
+
+sleep 5
+#docker exec bess-pfcpiface pfcpiface -config /conf/upf.jsonc -simulate create
```
[the remark this answers is not in the document]
Comment (sureshmarikkannu) [reply ptf/config/docker_setup.sh]: done

File: ptf/tests/linerate/baseline.py [hunk 4]
```diff
@@ -8,18 +8,20 @@
...
+TREX_SRC_MAC = "b4:96:91:b4:4b:09"
+#Destination MAC address for DL traffic
+UPF_DEST_MAC = "b4:96:91:b2:06:41"
 
 # Port setup
-TREX_SENDER_PORT = 0
-TREX_RECEIVER_PORT = 1
-BESS_SENDER_PORT = 2
-BESS_RECEIVER_PORT = 3
+TREX_SENDER_PORT = 1
```
[the remark this answers is not in the document]
Comment (sureshmarikkannu) [reply ptf/tests/linerate/baseline.py:21]: done

File: ptf/tests/linerate/baseline.py [hunk 5]
```diff
@@ -8,18 +8,20 @@
...
+TREX_RECEIVER_PORT = 0
+BESS_SENDER_PORT = 1
+BESS_RECEIVER_PORT = 0
```
[the remark this answers is not in the document]
Comment (sureshmarikkannu) [reply ptf/tests/linerate/baseline.py]: done

File: ptf/tests/linerate/qos_metrics.py [hunk 6]
```diff
@@ -11,13 +11,14 @@
...
+#Destination MAC address for DL traffic
+UPF_DEST_MAC = "b4:96:91:b2:06:41"
 
 # Port setup
-TREX_SENDER_PORT = 0
-TREX_RECEIVER_PORT = 1
-BESS_SENDER_PORT = 2
-BESS_RECEIVER_PORT = 3
+TREX_SENDER_PORT = 1
+TREX_RECEIVER_PORT = 0
```
[the remark this answers is not in the document]
Comment (sureshmarikkannu) [reply ptf/tests/linerate/qos_metrics.py]: done

File: ptf/README.md
[the remark this answers is not in the document]
Comment (sureshmarikkannu) [reply ptf/README.md:1]: i did mentioned it already in README as follows,

"* `upf/ptf/.env` file updated with `UPF_ADDR` and `TREX_ADDR` parameters"

[the remark this answers is not in the document]
Comment (gab-arrobo) [reply ptf/README.md:1]: Ok, then I overlooked at it. Thanks!

**Review by gab-arrobo — APPROVED** [review]
+1

**Pull request merged**